code/game/atoms.dm

```diff
@@ -246,12 +249,19 @@ its easier to just keep the beam vertical.
 	//user << "[name]: Dn:[density] dir:[dir] cont:[contents] icon:[icon] is:[icon_state] loc:[loc]"
 
 	if(reagents && is_open_container()) //is_open_container() isn't really the right proc for this, but w/e
-		user << "It contains:"
-		if(reagents.reagent_list.len)
-			for(var/datum/reagent/R in reagents.reagent_list)
-				user << "[R.volume] units of [R.name]"
+		if(get_dist(user, src) <= 1 || is_transparent_container())
+			user << "It contains:"
+			if(reagents.reagent_list.len)
+				var/obj/item/clothing/glasses/G = user.get_item_by_slot(slot_glasses)
+				if(G && (G.scan_vision & SCAN_CHEMICAL) && get_dist(user, src) <= 1)
+					for(var/datum/reagent/R in reagents.reagent_list)
+						user << "[R.volume] units of [R.name]"
+				else
+					user << "[reagents.total_volume] units of chemical" // Water is a chemical, right?
+			else
+				user << "Nothing."
 		else
-			user << "Nothing."
+			user << "You cannot see what it contain from here."

```

Comment:
Could just go with the same as paper.
"<span class='notice'>It is too far away.</span>"